Source/WebSocket.swift

```diff
@@ -1668,13 +1668,17 @@ public class WebSocket: NSObject {
     public init(request: NSURLRequest, subProtocols : [String] = []){
         opened = true
         ws = InnerWebSocket(request: request, subProtocols: subProtocols, stub: false)
+        super.init()
+        ws.eclose = { [unowned self] in
```

Comment:
@ricardopereira 
The empty NSURLRequest is used as a placeholder for the delayed connection feature #14. A delayed connection is simply a websocket that has does not know the endpoint until the `open()` method is called. 

The pattern looks something like this:

```swift
ws = WebSocket()   // Create websocket, but do not make a connection yet. 
ws.event.close = {
    ws.open(); // reopen the connection.
}
// then at some other point in time we open the connection with 
ws.open("ws://domain.com/endpoint") // Open 
```